### PR TITLE
ENH: Add butterfly channel picking

### LINKED ISSUE #2015: plot bad channel on top
Currently, the bad channel is sandwiched among the other lines in the butterfly plot. It should be on top of the plot. This can be hopefully done by setting `zorder=-1`

cc @Eric89GXL

### LINKED ISSUE #2071: identify channels in butterfly plot
It would be great if there would be a way to identify a bad channel in the topo butterfly plot. sometimes you just don't know how bad a channel is until you plot it among the others.

### PR DESCRIPTION
**Pull request by Eric89GXL**
Closes #2071.
Closes #2015.

Ready for testing / review / merge @teonlamont.

### CODE REVIEW

File: mne/source_estimate.py [hunk 1]
```diff
@@ -2289,7 +2292,7 @@ def compute_morph_matrix(subject_from, subject_to, vertices_from, vertices_to,
             continue
         m = sparse.eye(len(idx_use), len(idx_use), format='csr')
         morpher[hemi] = _morph_buffer(m, idx_use, e, smooth, n_vertices,
-                                      vertices_to[hemi], maps[hemi])
+                                      vertices_to[hemi], maps[hemi], warn=warn)
```
Comment (Eric89GXL) [inline mne/source_estimate.py]: This is an unrelated but very simple change to the morphing code to complete our use of the `warn` parameter. I can split into a separate PR if necessary, but this PR is only +65/-16 so hopefully it's not too bad.

**Comment by teonlamont** [comment]
it works really well. is there a way to make the text the top layer in maybe a different color? if you click in a place where there are surrounding, the text gets washed out. 
do we want it to have the behavior like `raw.plot()` where you click the channel, it's marked as bad and you toggle it on and off for multiple channels?

**Comment by teonlamont** [comment]
I got one error but it's when I clicked on the plot with no close by channels:

```python
In [11]: Traceback (most recent call last):
  File "/Users/teon/anaconda/lib/python2.7/site-packages/matplotlib/backends/backend_qt5.py", line 263, in mousePressEvent
    FigureCanvasBase.button_press_event(self, x, y, button)
  File "/Users/teon/anaconda/lib/python2.7/site-packages/matplotlib/backend_bases.py", line 1887, in button_press_event
    self.callbacks.process(s, mouseevent)
  File "/Users/teon/anaconda/lib/python2.7/site-packages/matplotlib/cbook.py", line 540, in process
    proxy(*args, **kwargs)
  File "/Users/teon/anaconda/lib/python2.7/site-packages/matplotlib/cbook.py", line 415, in __call__
    return mtd(*args, **kwargs)
  File "mne/viz/evoked.py", line 48, in _butterfly_on_button_press
    if params['need_draw']:
KeyError: 'need_draw'
```

**Comment by Eric89GXL** [comment]
Fixed the clicking bug and the text display, see if you're happy now.

I'm -1 on making this a bad channel setter. It is better to do bad channel selection at the raw stage since things like projector calculation and ICA will change depending on which channels are marked bad.

**Comment by agramfort** [comment]
works like a charm for me but travis complains 

thanks @Eric89GXL !

**Comment by teonlamont** [comment]
sweet. works great :)

**Comment by teonlamont** [comment]
thanks @Eric89GXL!

**Comment by mainakjas** [comment]
@jaeilepp do you want to participate in code review here? Seems relevant to the GSoC

**Comment by Eric89GXL** [comment]
@teonlamont in case you didn't notice I made it so that clicking away from all the channels removes the text.

Turns out that we weren't actually using old numpy/scipy before because subsequent calls to `conda install` were updating those packages. I made changes to `.travis.yml` fix it in this PR. Then I made some other scattered changes to fix for code for those old versions of numpy and scipy now that we actually test them.

Ready for review/merge.

File: .travis.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ env:
     - PYTHON=2.7 DEPS=full TEST_LOCATION=src
     - PYTHON=2.7 DEPS=nodata TEST_LOCATION=src MNE_DONTWRITE_HOME=true  # also runs flake8
     - PYTHON=3.4 DEPS=full TEST_LOCATION=install MNE_STIM_CHANNEL=STI101
-    - PYTHON=2.6 DEPS=full TEST_LOCATION=src NUMPY="=1.6" SCIPY="=0.11" MPL="=1.1" SKLEARN="=0.11" PANDAS="=0.8"
+    - PYTHON=2.6 DEPS=full TEST_LOCATION=src NUMPY="=1.7" SCIPY="=0.11" MPL="=1.1" SKLEARN="=0.11" PANDAS="=0.8"
```
Comment (agramfort) [inline .travis.yml]: you bumped the numpy version to 1.7?

**Comment by agramfort** [comment]
I'd like to make sure we're not silently bumping minimum numpy version but besides +1 for merge.

**Comment by Eric89GXL** [comment]
Had to bump it, 1.6 doesn't work (library error), if you want to see it
look at the Travis run before the version bump

**Comment by Eric89GXL** [comment]
Specifically:

https://travis-ci.org/mne-tools/mne-python/jobs/61931138#L176

**Comment by agramfort** [comment]
Bumping numpy version will have impact on Debian packaging :(

Is there a way to avoid it?

**Comment by Eric89GXL** [comment]
No no no, the version is only bumped *in Travis*, not as a requirement for the package. We have never run the oldest versions in Travis anyway because they are not available (e.g., scipy's oldest available is 0.11 but we only require something like 0.9).

**Comment by Eric89GXL** [comment]
...and we weren't actually even running numpy 1.6 before in Travis, we were using 1.8 or 1.9 because it got auto-installed by recommendations from later `conda install` commands.

**Comment by agramfort** [comment]
Ok good. Merge when Travis is happy

**Comment by Eric89GXL** [comment]
@agramfort I pushed a commit to address #2015, do you want to have another quick look or are you okay with it?

**Comment by Eric89GXL** [comment]
Specifically: https://github.com/mne-tools/mne-python/commit/236df09164402be6be0dcd754f7deaee0d097854

**Comment by agramfort** [comment]
Ok when Travis is happy

**Pull request merged**